src/flags/run_time_configurable.cpp

```diff
@@ -9,103 +9,170 @@
-    const auto server_name = memgraph::utils::global_settings.GetValue(kServerNameSettingKey);
-    MG_ASSERT(server_name, "Bolt server name is missing from the settings");
-    *(bolt_server_name_.Lock()) = *server_name;
-  });
-  // Update value from read settings
-  const auto &name = memgraph::utils::global_settings.GetValue(kServerNameSettingKey);
-  MG_ASSERT(name, "Failed to read server name from settings.");
-  *(bolt_server_name_.Lock()) = *name;
-  // Override server name if passed via command line argument
-  if (!FLAGS_bolt_server_name_for_init.empty()) {
-    memgraph::utils::global_settings.SetValue(kServerNameSettingKey, FLAGS_bolt_server_name_for_init);
+auto ToLLEnum(std::string_view val) {
+  const auto ll_enum = memgraph::flags::LogLevelToEnum(val);
+  if (!ll_enum) {
+    throw memgraph::utils::BasicException("Unsupported log level {}", val);
   }
+  return *ll_enum;
+}
 
-  // Register query timeout
-  memgraph::utils::global_settings.RegisterSetting(kQueryTxSettingKey, kDefaultQueryTx, [&] {
-    const auto query_tx = memgraph::utils::global_settings.GetValue(kQueryTxSettingKey);
-    MG_ASSERT(query_tx, "Query timeout is missing from the settings");
-    execution_timeout_sec_ = std::stod(*query_tx);
-  });
-  // Update value from read settings
-  const auto &tx = memgraph::utils::global_settings.GetValue(kQueryTxSettingKey);
-  MG_ASSERT(tx, "Failed to read query timeout from settings.");
-  execution_timeout_sec_ = std::stod(*tx);
-  // Override query timeout if passed via command line argument
-  if (FLAGS_query_execution_timeout_sec != -1) {
-    memgraph::utils::global_settings.SetValue(kQueryTxSettingKey, std::to_string(FLAGS_query_execution_timeout_sec));
-  }
+bool ValidBoolStr(std::string_view in) {
+  const auto lc = memgraph::utils::ToLowerCase(in);
+  return lc == "false" || lc == "true";
+}
+
+auto GenHandler(std::string flag, std::string key) {
+  return [key = std::move(key), flag = std::move(flag)]() -> std::string {
+    const auto &val = memgraph::utils::global_settings.GetValue(key);
+    MG_ASSERT(val, "Failed to read value at '{}' from settings.", key);
+    gflags::SetCommandLineOption(flag.c_str(), val->c_str());
+    return *val;
+  };
+}
+
+}  // namespace
+
+namespace memgraph::flags::run_time {
 
-  // Register log level
-  auto get_global_log_level = []() {
-    const auto log_level = memgraph::utils::global_settings.GetValue(kLogLevelSettingKey);
-    MG_ASSERT(log_level, "Log level is missing from the settings");
-    const auto ll_enum = memgraph::flags::LogLevelToEnum(*log_level);
-    if (!ll_enum) {
-      throw utils::BasicException("Unsupported log level {}", *log_level);
+void Initialize() {
+  constexpr bool kRestore = true;
+  auto register_flag = [&](
+                           const std::string &flag, const std::string &key, bool restore,
+                           std::function<void(const std::string &)> post_update = [](auto) {},
```

Comment:
For avoiding allocations prefer `auto &&` to take the lambda, rather than `std::function`

```cpp
#include <concepts>

auto demo(std::invocable<int> auto && f)
{ 
    return f(42);
}

int main()
{
    return demo([](int x){ return x; });
}
```

Reply:
Unfortunately I am not sure you can have a default post_update this way.
Am I wrong?

Reply:
You are right you would need to split the overload set by introducing a new overload 
```cpp
#include <concepts>

auto demo(std::invocable<int> auto && f)
{ 
    return f(2);
}

inline auto demo()
{
    return demo(/*default =*/[](int){ return 40;});
}

int main()
{
    return demo() + demo([](int x){ return x; });
}
```

Reply:
I would leave it as is. 
This is an initialization of run time settings and I would rather have a single function implementation than better performance.